packages/engine/src/behave-graph/components/RuntimeGraphComponent.ts

```diff
@@ -0,0 +1,36 @@
+import {
+  DefaultLogger,
+  Engine,
+  ManualLifecycleEventEmitter,
+  readGraphFromJSON,
+  registerCoreProfile,
+  Registry
+} from 'behave-graph'
+
+import { getState } from '@xrengine/hyperflux'
+
+import { defineComponent, getComponent } from '../../ecs/functions/ComponentFunctions'
+import { BehaveGraphSystemState } from '../systems/BehaveGraphSystem'
+import { BehaveGraphComponent } from './BehaveGraphComponent'
+
+export const RuntimeGraphComponent = defineComponent({
+  name: 'EE_runtimeGraph',
+
+  onInit: (entity) => {
+    const graphComponent = getComponent(entity, BehaveGraphComponent)
+    const registry = new Registry()
+    const logger = new DefaultLogger()
+    const ticker = new ManualLifecycleEventEmitter()
+    registerCoreProfile(registry, logger, ticker)
+    const systemState = getState(BehaveGraphSystemState)
+    systemState.domains[graphComponent.domain].value.register(registry, logger, ticker)
```

Comment:
components should be thought of as just data,data structures and must not have side effects like this, we should put all of this in an enter query or useEffect

Reply:
None of this is side effecting, lines 24 and 26 only modify the registry argument they are passed.
Also, the Engine and ManualLifecycleEmitter types have no default value, so I'd need to be returning either nulls (bad) or unsafely cast dud objects (also bad). The Engine constructor must accept a fully initialized Graph, which requires a registry that is staged using the registerCoreProfile function and whatever additional domain-specific, also not side effecting register function that is stored in the systemState.domains record.